src/configure

```diff
@@ -10828,7 +10893,7 @@ printf "%s\n" "#define HYPRE_USING_HIP 1" >>confdefs.h
            HYPRE_CUDA_GENCODE="`echo ${HYPRE_CUDA_GENCODE}|sed 's/,$//'`"
         fi
 
-                                        HIPCXXFLAGS="-x hip -std=c++14 ${HIPCXXFLAGS}"
+                                        HIPCXXFLAGS="-x hip -fPIE -std=c++14 ${HIPCXXFLAGS}"
```

Comment:
This change breaks the shared build which is default in Spack where I noticed the issue. One possible solution is to swap the order of `SHARED_COMPILE_FLAG` and `HIPCXXFLAGS` below when re-defining `HIPCXXFLAGS` for the shared build -- this way `-fPIC` will come after `-fPIE` and overwrite it, I think.

ping: @victorapm

Reply:
Thank you for pointing out @v-dobrev. I've just fixed this on master

Reply:
Great! Thanks!